include/flang/Semantics/symbol.h

```diff
@@ -91,12 +92,15 @@ ENUM_CLASS(SubprogramKind, Module, Internal)
 // type information.
 class SubprogramNameDetails {
 public:
-  SubprogramNameDetails(SubprogramKind kind) : kind_{kind} {}
+  SubprogramNameDetails(SubprogramKind kind, ProgramTree &node)
+    : kind_{kind}, node_{node} {}
   SubprogramNameDetails() = delete;
   SubprogramKind kind() const { return kind_; }
+  ProgramTree &node() const { return *node_; }
```

Comment:
`ProgramTree` nodes only exist during name resolution but `Symbol`s live longer. So I think this will be a dangling reference if it is called after name resolution.

Reply:
Sure, but there shouldn't be any SubprogramNameDetails symbols after name resolution, right?

Reply:
Oh, right.